### ISSUE TITLE
Not showing existing tags for typescript

### ISSUE THREAD

**Issue by ayroblu** [majutsushi/tagbar#596]
Hi, I'm not sure what's going on, but I managed to get it to work for `python`, but on my `typescript` files I get nothing. This is my debug output:

![image](https://user-images.githubusercontent.com/4915682/77236693-ec401a80-6bb8-11ea-8b08-611d7b89fb33.png)

I'm not sure what to do, but I get this empty tagbar:

![image](https://user-images.githubusercontent.com/4915682/77236716-0bd74300-6bb9-11ea-91ff-355b9095b856.png)

My related vim setup:
![image](https://user-images.githubusercontent.com/4915682/77236773-82744080-6bb9-11ea-87cb-ae55f211e5a1.png)

UPDATE:
If I use the tstags option in the wiki as opposed to the ctags option, open vi without any ts files first, then open my ts file, then it shows something, but if I open it directly it doesn't work plus shows an error on new files, which is not great

**Comment by alerque** [comment]
Have you tried it with no manual configuration at all? A lot of stuff in the Wiki is outdated as UCtags has started supporting a lot more things than before and we've been updating the out of the box configurations.

**Comment by ayroblu** [comment]
Yes, I tried it without config first then was trying to get it to work. Note that I’m using exuberant ctags. (I briefly tried uctags but found it a bit broken for my existing personal setup so decided to stick with ectags)

**Comment by alerque** [comment]
@ayroblu Note that Exuberant ctags is no longer supported. We've tried not to actively remove compatibility with it, but it's over a decade long in the tooth now and we _know_ it doesn't work for a lot of things. I'm happy to accept a PR to fix it, but there is no way I'm going to muck around with installing an antiquated ctags version to add support for a modern language. I'm not saying it can't be made to work, just that I'm not going to spend any time supporting that. I'd be much happier to use whatever slivers of time I get to support universal ctags.

**Comment by alerque** [comment]
I'm closing this because it's not something we're tracking with intention of ever fixing. That being said if somebody does want to fix it and add support we wouldn't necessary reject a PR. Meanwhile we strongly recommend Universal Ctags and issue reports for any problems involving it.

**Comment by AdrienLemaire** [comment]
@alerque I'm also facing the same issue of not seeing any tag in my typescript file

Tried with both the universal ctags setup and tstags.js setup.

I can confirm that running tstags on my file returns results:
![](https://i.imgur.com/wqih7O5.png)

```console
$ sudo npm install --global git+https://github.com/Perlence/tstags.git 
```

Config:
```vim
Plug 'majutsushi/tagbar'
let g:tagbar_type_typescript = {                                                  
  \ 'ctagsbin' : 'tstags',                                                        
  \ 'ctagsargs' : '-f-',                                                           
  \ 'kinds': [                                                                     
    \ 'e:enums:0:1',                                                               
    \ 'f:function:0:1',                                                            
    \ 't:typealias:0:1',                                                           
    \ 'M:Module:0:1',                                                              
    \ 'I:import:0:1',                                                              
    \ 'i:interface:0:1',                                                           
    \ 'C:class:0:1',                                                               
    \ 'm:method:0:1',                                                              
    \ 'p:property:0:1',                                                            
    \ 'v:variable:0:1',                                                            
    \ 'c:const:0:1',                                                              
  \ ],                                                                            
  \ 'sort' : 0                                                                    
\ }

noremap <leader>tt :TagbarToggle<CR>
noremap <leader>tj :TagbarOpen j<CR>
```

Any idea why?

**Comment by alerque** [comment]
You don't need a custom tag def any more. If your plugin is up to date just remove that bit of config…and maybe fix the plugin location while you are at it.

```viml
Plug 'preservim/tagbar​'​

​noremap​ ​<leader>​tt :TagbarToggle​<CR>​
​noremap​ ​<leader>​tj​ :TagbarOpen ​j​<CR>
```

**Comment by AdrienLemaire** [comment]
@alerque thanks! And I also needed to remove the typescript related language definition from the `~/.ctags` file, now it works fine :tada: 
Maybe the readme/wiki could be update to better explain this :)

**Comment by alerque** [comment]
Built in TypeScript support just landed yesterday in #669 thanks to @joshfrench. It would be great if those of you who actually use TypeScript could update the wiki (which is free for anyone to edit) and the README (which you can edit here on GitHub and submit as a PR) with up to date documentation that works. I would keep a mention of TS even though it doesn't need extra configuration, but maybe fill in under the heading this information about what should be removed.

**Comment by joshfrench** [comment]
I just updated the wiki. @AdrienLemaire if possible, use `u-ctags` instead of `tstags`; `tstags` is significantly less precise. If you don't want to switch bins, you'll need to keep your config as above.
